Highway.cpp

```diff
@@ -13,16 +17,39 @@ void Highway::changeSpeed(int newSpeed)
 
 void Highway::addVehicleInternal(Vehicle* v)
 {
-    assert(false);
-
+    // assert(false);
+    if(auto * car = dynamic_cast<Car *>(v))
+    {
+        car->closeWindows();
+    }
+    if(auto * bike = dynamic_cast<Motorcycle *>(v))
+    {
+        bike->lanesplitAndRace(120);
+    }
+    if(auto * truck = dynamic_cast<SemiTruck *>(v))
+    {
+        truck->honkForSpace();
+    }
```

Comment:
use `else if`
you're still checking if it's a `Motorcycle*` even if it's a `Car*`
- [ ] revise